### ISSUE TITLE
Alphabetize entries inside groups

### ISSUE THREAD

**Issue by marklarr** [venmo/synx#26]
It's annoying how entries aren't forced into alphabetization in Xcode like they are on the file system. There is a 'sort by name' option if you right click on a folder -- it'd be cool to automatically do this.

**Comment by eytanbiala** [comment]
+1 !

**Comment by foobra** [comment]
+1

**Comment by ScottRobbins** [comment]
I had mentioned this with @marklarr on twitter, but it would be great to get an option to not alphabetize with synx (or have alphabetizing as a flag to enable).

Having this will just be a little difficult with a team and longstanding project. Running it will reorganize a lot of the project file, which may cause issues and not everyone on my team is going to run synx (and surely won't alphabetize the files). 

Synx works really well for me because I can organize my Xcode group however I want, and it will organize the file system which I never really wanted to manually touch. But forcing alphabetization doesn't follow this same pattern.

**Comment by FiRiN59** [comment]
Please make sorting optional and switched off by default! Folder structure is one thing, but logical project organization is sacred for many of us. 

P.S. Ideally you should have different options for groups and files.

**Comment by marklarr** [comment]
I asked around to the team at work, because I wanted to verify that it was a safe assumption that users of Synx would be a fan of Xcode mirroring the file system as closely as possible, but apparently we all assumed incorrectly :laughing:. Sorry to anyone that this has gotten in the way of -- the good news is, you can lock-in to `synx v0.0.51` until I can rollout a flag for this. Honestly, unless the community as a whole feels strongly otherwise, I'd like to keep the flag on by default.

**Comment by FiRiN59** [comment]
You mean v0.0.61, right?

As for the flag... For me synx is a great tool that takes my project's logical structure and tries to mimic it on the disk as closely as possible. I don't expect it to alter the project itself. Another consideration is backward compatibility. You are adding a feature that changes default behaviour and that's not a good thing for existing users. So my vote is: "It's good to have this feature, but it should be off by default." But of course the tool is in 0.1.x stage, so for the long term vision "default on" might make more sense.

**Comment by marklarr** [comment]
Yes sorry, `v 0.0.61`, thanks for catching that.

Synx already alters the project itself -- you'll notice that, after running Synx on a disorganized project, the `.pbxproj` will have a lot of diff. As for backwards compatibility, people can just shut the flag off :smile:

**Comment by FiRiN59** [comment]
I don't want to be closer to the disk, I want the disk to be closer to me (logical structure of the project). But anyways it's a great tool and I can live with "default on".

**Comment by ScottRobbins** [comment]
Yeah I agree with @FiRiN59 , but I won't complain as it's a great tool that you've made for free :P I appreciate you considering adding a flag to not alphabetize

**Comment by AndrewSB** [comment]
I agree with @ScottRobbins & @FiRiN59, having a flag that turns off Alphabetizing would be awesome. Having my project order restructured is not fun

**Comment by rafalwojcik** [comment]
@AndrewSB @FiRiN59 @marklarr @eytanbiala @foobra Done :)

**Comment by lukaskubanek** [comment]
@rafalwojcik Great job, thank you! I've been silently waiting for this feature for so long.

**Comment by AndrewSB** [comment]
@rafalwojcik Made my morning 😊

**Comment by FiRiN59** [comment]
@rafalwojcik Amazing, thx!

**Comment by BluMist** [comment]
This is a tad nitpicky, but the sort order is different from what XCode does when "Sort By Name". 

I believe the sort order is case insensitive in XCode, space/symbol before alphanumeric; just a couple inconsistencies that I have noticed.

**Comment by ScottRobbins** [comment]
Very late, but @rafalwojcik You're a wonderful person.
